jaeger-thrift/src/test/java/io/jaegertracing/thrift/internal/senders/UdpSenderTest.java

```diff
@@ -84,21 +84,29 @@ public void tearDown() throws Exception {
     reporter.close();
   }
 
-  @Test(expected = SenderException.class)
-  public void testAppendSpanTooLarge() throws Exception {
+  /**
+   * create a mock span with target size
+   */
+  private JaegerSpan buildSpanWithSize(int targetSpanSize) throws Exception {
     JaegerSpan jaegerSpan = tracer.buildSpan("raza").start();
-    String msg = "";
-    for (int i = 0; i < 10001; i++) {
-      msg += ".";
-      jaegerSpan.log(msg);
-    }
+    jaegerSpan.setTag("mock", "");
+    io.jaegertracing.thriftjava.Span emptySpan = JaegerThriftSpanConverter.convertSpan(jaegerSpan);
 
-    try {
-      sender.append(jaegerSpan);
-    } catch (SenderException e) {
-      assertEquals(e.getDroppedSpanCount(), 1);
-      throw e;
-    }
+    int emptySpanSize = sender.getSize(emptySpan);
+    int freePacketSizeLeft = targetSpanSize - emptySpanSize;
+
+    // each "0" takes 1 byte in UTF-8
+    // size header seems to take 1 byte for every 2^7 string length
+    int lengthBytes = (31 - Integer.numberOfLeadingZeros(freePacketSizeLeft)) / 7;
```

Comment:
Could you please explain better in comments why 31 and 7?

Reply:
`(31 - Integer.numberOfLeadingZeros(x))` is a general formula for calculating stable log base 2 (w/o using floating point)
`7` is just a number I found but I don't understand how the protocol takes additional one byte every 2^7 character. I even read the protobuf spec, but don't know (why 7 instead of 8).
Overall I think the code is quite cryptic, do you think I should try a different approach? (maybe brute force by subtracting '0's one by one until `targetSpanSize == spanSize`)